src/box/lua/connpool.lua

```diff
@@ -271,46 +316,102 @@ local function filter(opts)
             table.insert(static_candidates, instance_name)
         end
     end
-    -- Return if retrieving dynamic information is not required.
-    if next(static_candidates) == nil or next(dynamic_opts) == nil then
+
+    if mode == 'any' then
```

Comment:
The comment seems useful, especially because a general 'if there are dynamic opts' check now looks as a `mode` check. Maybe is even worth to explicitly define some `has_dynamic_filters` predicate.

Why the `static_candidates` check for emptiness is removed?

Reply:
Thank you. 

I've added the comment line on skipping connection phase if the dynamic props not needed back. 

For some reason I though that `acquire_all_candidates()` will just pass if it receives an empty array. It didn't work so. 

I've left the `static_candidates` emptiness check removed but made `acquire_all_candidates()` just pass if it receives an empty array of instances. It seems for me this is a little bit more consistent.